lib/table.c

```diff
@@ -391,7 +391,7 @@ void route_node_delete(struct route_node *node)
 struct route_node *route_top(struct route_table *table)
 {
 	/* If there is no node in the routing table return NULL. */
-	if (table->top == NULL)
+	if (table == NULL || table->top == NULL)
```

Comment:
I'm a bit hesitant to assume that the table will be non-null, this seems like a benign change that just shifts the crash to elsewhere.  Why is the table not created?